### PR TITLE
Add user setting to turn off most animations

### PR DESCRIPTION
**Pull request by MetRonnie**
The main benefit to this other than for accessibility concerns is reduced latency on virtual desktops.

This PR also improves the mutation menu appearance after #1269

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests are included
- [x] `CHANGES.md` entry included if this is a change that can affect users

### CODE REVIEW

File: src/components/cylc/cylcObject/Menu.vue [hunk 1]
```diff
@@ -28,13 +28,15 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-        max-width="600px"
+        width="max-content"
+        max-width="min(600px, 100%)"
         theme="dark"
         position="absolute"
         :style="{
           left: `${x}px`,
-          top: `${y}px`
+          top: `${y}px`,
+          // Set transition origin relative to clicked target:
```
Comment (oliver-sanders) [inline src/components/cylc/cylcObject/Menu.vue:38]: I think the clicked target could be on any side?

File: src/mixins/appSettings.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export default {
```
Comment (oliver-sanders) [inline src/mixins/appSettings.js:18]: Is this mixin needed? Can this code go in [src/store/app.module.js](https://github.com/cylc/cylc-ui/pull/1330/files#diff-d7b6802e371ab4383805859b39da1ae0320e4a80540d287cfbcc9db88c84e152)?

File: tests/e2e/specs/userprofile.cy.js [hunk 3]
```diff
@@ -89,6 +89,44 @@ describe('User Profile', () => {
...
+        .then(([$el]) => {
+          animationSpy = cy.spy($el, 'animate')
+        })
+        .get('.c-interactive:first')
+        .click()
+        .get('@mutationMenu')
+        .should('be.visible')
+        .then(() => {
+          if (shouldAnimate) {
+            expect(animationSpy).to.be.called
```
Comment (oliver-sanders) [inline tests/e2e/specs/userprofile.cy.js:108]: Nice!

**Review by wxtim — APPROVED** [review]
Nothing to say beyond what @oliver-sanders  has said. 😄

File: src/mixins/appSettings.js (hunk 2 above)
Comment (MetRonnie) [reply src/mixins/appSettings.js:18]: I looked into making this a Vuex action but it needs access to `this.$vuetify` on the component instance

Comment (oliver-sanders) [reply src/mixins/appSettings.js:18]: Dammit! We should really collect these options, they are all managed differently ATM, but this suggests VueX isn't the place to collect them.

**Review by oliver-sanders — APPROVED** [review]
:1st_place_medal:

**Comment by oliver-sanders** [comment]
Works really nicely, should remember to announce this feature when we deploy.

**Pull request merged**